app/models/scheme.rb

```diff
@@ -17,4 +17,13 @@ class Scheme < ApplicationRecord
 
   include PublicActivity::Model
   tracked owner: ->(controller, _) { controller.current_user if controller }
+
+  def set_start_date(date)
```

Comment:
Some similar code is in use in a couple of places now. Can we make this reusable somehow?